### DISCUSSION TITLE
Build on aarch64 / linux fails

### DISCUSSION THREAD

**Discussion by tosttost** [coin-or/Cbc#361] [Discussion]
Hello,

has anyone managed to build Cbc for linux on aarch64 (e.g. raspberry pi with 64bit OS, mobile phones that run some favor of linux instead of android, ...)?

I tried to use coinbrew and configure tells me that I should try update config.guess and claims a timestamp of 2007-05-17 i.e. long before the birth of aarch64. If that fails sending a patch to the gnu project, but I'm pretty sure recent autotools support aarch64 ;)

Is there someone who knows autotools and can update does files? My experience is limited to "./configure && make"...

**Comment by tosttost** [comment]
It looks like this was fixed in master: tell coinbrew to build master instead of the latest stable branch and it works.

**Comment by tkralphs** [comment]
As you discovered, the version of the COIN-OR [BuildTools](https://github.com/coin-or-tools/BuildTools) that is used by the `stable/2.10` stack is based on an old version of the autotools, which we were stuck on for years because we had a number of fixes and patches accumulated over a long time that only worked in that specific old version. Moving back to current autotools required a a massive overhaul of the entire build system, which has now been (mostly) completed (a few things are still being improved, see coin-or-tools/BuildTools#148). So yes, the solution is to move to `master`, but that comes of course with the attendant possible instability of a development version. Ultimately, `master` should be a much more capable and robust version of Cbc than `stable/2.10`,  but there may still be a few growing pains ahead yet, as we do some refactoring.
